diff --git a/board/atb/atb_rk3568j_smc_r1/genimage-atb_rk3568j_smc_r1.cfg b/board/atb/atb_rk3568j_smc_r1/genimage-atb_rk3568j_smc_r1.cfg
index 2e3c77fd9f..ec40f275e2 100644
--- a/board/atb/atb_rk3568j_smc_r1/genimage-atb_rk3568j_smc_r1.cfg
+++ b/board/atb/atb_rk3568j_smc_r1/genimage-atb_rk3568j_smc_r1.cfg
@@ -1,17 +1,4 @@
 
-image boot.vfat {
-	vfat {
-		label = "boot"
-		files = {
-			"uboot.env",
-			"linux",
-			"dtb",
-			"rootfs"
-		}
-	}
-	size = 512M
-}
-
 image sdcard.img {
 	hdimage {
 	}
@@ -27,16 +14,4 @@ image sdcard.img {
 		image = "uboot.img"
 		offset = 8M
 	}
-
-	partition boot {
-		partition-type = 0xC
-		bootable = "true"
-		image = "boot.vfat"
-		offset = 16M
-	}
-
-	partition rootfs {
-		partition-type = 0x83
-		image = "ext.rootfs.ext4"
-	}
 }
diff --git a/board/atb/atb_rk3568j_smc_r1/post-image.sh b/board/atb/atb_rk3568j_smc_r1/post-image.sh
index 5d94c86b1f..6045425a0c 100755
--- a/board/atb/atb_rk3568j_smc_r1/post-image.sh
+++ b/board/atb/atb_rk3568j_smc_r1/post-image.sh
@@ -17,7 +17,7 @@ SOC=RK3568J
 BOARD_DIR="$(dirname $0)"
 BOARD_NAME="$(basename ${BOARD_DIR})"
 GENIMAGE_CFG="${BOARD_DIR}/genimage-${BOARD_NAME}.cfg"
-IMAGE_NAME="$(basename -s .dtb $2)-usd"
+IMAGE_NAME="$(basename -s .dtb $2)-spinor"
 
 echo "argc = $#"
 echo "arg0 = $0"
@@ -109,65 +109,6 @@ fi
 cp u-boot.itb ${IMAGES}/uboot.img
 cd -
 
-# 3. Linux kernel
-cp ${IMAGES}/Image ${IMAGES}/linux
-
-# 4. Linux device tree blob
-cp ${IMAGES}/${DTB} ${IMAGES}/dtb
-
-# 5. Rootfs
-cp ${IMAGES}/rootfs.cpio.gz ${IMAGES}/rootfs
-${BUILD}/uboot-${UBOOT_REPO}/tools/mkimage -A arm -T ramdisk -C gzip -d ${OUTPUT}/images/rootfs.cpio.gz ${OUTPUT}/images/rootfs
-
-# These images are stored in dl directory where all packages being used for building are stored.
-#ROOTFS_IMG="debian10-lxde.rootfs.ext4"
-#ROOTFS_IMG="debian10-xfce4.rootfs.ext4"
-#ROOTFS_IMG="ubuntu18.04-lxde.rootfs.ext4"
-#ROOTFS_IMG="ubuntu20.04-lxqt.rootfs.ext4"
-ROOTFS_IMG="ubuntu20.04-minimal.rootfs.ext4"
-
-echo "External rootfs is ${ROOTFS_IMG}"
-
-if ! [ -e ${BUILDROOT_HOME}/dl/${ROOTFS_IMG} ]; then
-	cd ${BUILDROOT_HOME}/dl/
-	wget -T 1 --ftp-user='atbftp_user' --ftp-password='32Vj_hy%c@gR' ftp://ftp.atb-e.ru:2121/ATB_FTP/buildroot/${ROOTFS_IMG}
-	cd -
-fi
-
-# Copy new or replace existing image to avoid impact of changes made earlier
-cp -f ${BUILDROOT_HOME}/dl/${ROOTFS_IMG} ${IMAGES}/ext.rootfs.ext4
-
-echo ""
-echo ""
-echo "WARNING!"
-echo "The following operations require privileged access."
-echo "To be assured that nothing dangerous is executed, you can observe the following file $0"
-echo ""
-echo ""
-# Unmount and remove mnt directory if it exists by any reasons
-mountpoint ${IMAGES}/mnt -q
-if [ $? == 0 ]; then
-	sudo umount ${IMAGES}/mnt
-fi
-sudo rm -rf ${IMAGES}/mnt
-
-mkdir ${IMAGES}/mnt
-sudo mount ${IMAGES}/ext.rootfs.ext4 ${IMAGES}/mnt
-if ! [ $? == 0 ]; then
-	exit 2
-fi
-
-sudo cp ${TARGET_DIR}/etc/fstab ${IMAGES}/mnt/etc/
-sudo cp -r ${TARGET_DIR}/lib/modules ${IMAGES}/mnt/lib/
-
-sudo umount ${IMAGES}/mnt
-sudo rm -rf ${IMAGES}/mnt
-
-# Pass an empty rootpath. genimage makes a full copy of the given rootpath to
-# ${GENIMAGE_TMP}/root so passing TARGET_DIR would be a waste of time and disk
-# space. We don't rely on genimage to build the rootfs image, just to insert a
-# pre-built one in the disk image.
-
 UBOOT_ENV_SIZE=0x8000
 ${BUILD}/uboot-${UBOOT_REPO}/tools/mkenvimage -s ${UBOOT_ENV_SIZE} -o ${IMAGES}/uboot.env ${BOARD_DIR}/uboot/uboot.env.txt
 
