diff --git a/board/atb/atb_rk3568j_smc_r1/genimage-atb_rk3568j_smc_r1.cfg b/board/atb/atb_rk3568j_smc_r1/genimage-atb_rk3568j_smc_r1.cfg
index e0a1415d9a..4627b53a3e 100644
--- a/board/atb/atb_rk3568j_smc_r1/genimage-atb_rk3568j_smc_r1.cfg
+++ b/board/atb/atb_rk3568j_smc_r1/genimage-atb_rk3568j_smc_r1.cfg
@@ -1,42 +1,37 @@
 
-image recovery {
+image uboot {
 	vfat {
-		label = "recovery"
+		label = "uboot"
 		files = {
 			"uboot.env",
-			"linux",
 			"dtb",
-			"rootfs"
 		}
 	}
-	size = 512M
+	size = 1M
 }
 
 image sdcard.img {
-	hdimage {
-		partition-table-type = "gpt"
-		gpt-location = 1M
-	}
+
 	partition rk0-boot {
 		in-partition-table = false
 		image = "idblock.bin"
 		offset = 32k
 	}
 
-	partition rk1-boot {
-		in-partition-table = false
-		image = "uboot.img"
-		offset = 8M
+	hdimage {
+		partition-table-type = "gpt"
+		gpt-location = 1M
 	}
 
 	partition boot {
 		in-partition-table = true
-		image = "recovery"
-		offset = 16M
+		image = "uboot"
+		offset = 2M
 	}
 
-	partition rootfs {
-		in-partition-table = true
-		image = "ext.rootfs.ext4"
+	partition rk1-boot {
+		in-partition-table = false
+		image = "uboot.img"
+		offset = 8M
 	}
 }
diff --git a/board/atb/atb_rk3568j_smc_r1/post-image.sh b/board/atb/atb_rk3568j_smc_r1/post-image.sh
index 1647474b43..211a00d41c 100755
--- a/board/atb/atb_rk3568j_smc_r1/post-image.sh
+++ b/board/atb/atb_rk3568j_smc_r1/post-image.sh
@@ -17,7 +17,7 @@ SOC=RK3568J
 BOARD_DIR="$(dirname $0)"
 BOARD_NAME="$(basename ${BOARD_DIR})"
 GENIMAGE_CFG="${BOARD_DIR}/genimage-${BOARD_NAME}.cfg"
-IMAGE_NAME="$(basename -s .dtb $2)-usd"
+IMAGE_NAME="$(basename -s .dtb $2)-spinor"
 
 echo "argc = $#"
 echo "arg0 = $0"
diff --git a/board/atb/atb_rk3568j_smc_r1/uboot/atb_rk3568j_smc_r1_defconfig b/board/atb/atb_rk3568j_smc_r1/uboot/atb_rk3568j_smc_r1_defconfig
index 20a44ebddc..feb8a8be09 100644
--- a/board/atb/atb_rk3568j_smc_r1/uboot/atb_rk3568j_smc_r1_defconfig
+++ b/board/atb/atb_rk3568j_smc_r1/uboot/atb_rk3568j_smc_r1_defconfig
@@ -84,9 +84,6 @@ CONFIG_EFI_PARTITION_ENTRIES_NUMBERS=64
 CONFIG_SPL_OF_CONTROL=y
 CONFIG_OF_LIVE=y
 CONFIG_OF_SPL_REMOVE_PROPS="clock-names interrupt-parent assigned-clocks assigned-clock-rates assigned-clock-parents"
-CONFIG_ENV_IS_IN_FAT=y
-CONFIG_ENV_FAT_INTERFACE="mmc"
-CONFIG_ENV_FAT_DEVICE_AND_PART="1:1"
 # CONFIG_NET_TFTP_VARS is not set
 CONFIG_REGMAP=y
 CONFIG_SPL_REGMAP=y
