diff --git a/drivers/mmc/rockchip_dw_mmc.c b/drivers/mmc/rockchip_dw_mmc.c
index 1287551bf1..a78cb39c7f 100644
--- a/drivers/mmc/rockchip_dw_mmc.c
+++ b/drivers/mmc/rockchip_dw_mmc.c
@@ -42,7 +42,7 @@ int board_mmc_dm_reinit(struct udevice *dev)
 {
 	struct rockchip_dwmmc_priv *priv = dev_get_priv(dev);
 
-	if (!priv || !&priv->clk)
+	if (!priv)
 		return 0;
 
 	if (!memcmp(dev->name, "dwmmc", strlen("dwmmc")))
